### PR TITLE
[detekt] Add `parallel` and `target` options

### LINKED ISSUE #1958: [detekt] Support `--parallel` option
detekt has the `--parallel` option:

```
$ detekt --help
...
    --parallel
      Enables parallel compilation and analysis of source files. Do some
      benchmarks first before enabling this flag. Heuristics show performance
      benefits starting from 2000 lines of Kotlin code.
      Default: false
...
```

We could get the benefit if supporting it.

### PR DESCRIPTION
**Pull request by ybiquitous**
> Explain a summary, purpose, or background for this change.

This change adds the new options: `parallel` (disabled by default) and `target` (alias for `input`).

> Link related issues or pull requests.

Close #1958
Part of #2096 (`target` option)

> Check the following if needed.

- [x] Add a new entry to [CHANGELOG.md](https://github.com/sider/runners/blob/master/CHANGELOG.md) if this change is notable.

### CODE REVIEW

File: lib/runners/processor/detekt.rb [hunk 1]
```diff
@@ -60,14 +63,15 @@ def analyze(changes)
...
-        *cli_report,
+        "--report", "xml:#{report_file}",
+        *(config_linter[:baseline].then { |value| value ? ["--baseline", value] : [] }),
+        *(comma_separated_list(config_linter[:config]).then { |list| list ? ["--config", list] : [] }),
+        *(comma_separated_list(config_linter[:"config-resource"]).then { |list| list ? ["--config-resource", list] : [] }),
+        *(config_linter[:"disable-default-rulesets"] ? ["--disable-default-rulesets"] : []),
+        *(config_linter[:parallel] ? ["--parallel"] : []),
+        *(comma_separated_list(config_linter[:excludes]).then { |list| list ? ["--excludes", list] : [] }),
+        *(comma_separated_list(config_linter[:includes]).then { |list| list ? ["--includes", list] : [] }),
+        *(comma_separated_list(config_linter[:input]).then { |list| list ? ["--input", list] : []}),
```
Comment (ybiquitous) [inline lib/runners/processor/detekt.rb:74]: [note] This refactored for simplicity (inlining methods).

File: lib/runners/processor/detekt.rb [hunk 2]
```diff
@@ -60,14 +66,15 @@ def analyze(changes)
...
-        *cli_report,
+        "--report", "xml:#{report_file}",
+        *(config_linter[:baseline].then { |value| value ? ["--baseline", value] : [] }),
+        *(comma_separated_list(config_linter[:config]).then { |list| list ? ["--config", list] : [] }),
+        *(comma_separated_list(config_linter[:"config-resource"]).then { |list| list ? ["--config-resource", list] : [] }),
+        *(config_linter[:"disable-default-rulesets"] ? ["--disable-default-rulesets"] : []),
+        *(config_linter[:parallel] ? ["--parallel"] : []),
+        *(comma_separated_list(config_linter[:excludes]).then { |list| list ? ["--excludes", list] : [] }),
+        *(comma_separated_list(config_linter[:includes]).then { |list| list ? ["--includes", list] : [] }),
+        *(comma_separated_list(config_linter[:target] || config_linter[:input]).then { |list| list ? ["--input", list] : []}),
```
Comment (ybiquitous) [inline lib/runners/processor/detekt.rb:77]: [note] `target` is preffered over `input`.

**Comment by ybiquitous** [comment]
@yykamei Could you please review this if you have time?

**Review by yykamei — APPROVED** [review]
LGTM

**Comment by ybiquitous** [comment]
Thank you. 👍

**Pull request merged**